zephyr/include/rtos/string_macro.h

```diff
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2024 Intel Corporation. All rights reserved.
+ *
+ * Author: Baofeng Tian <Baofeng.tian@intel.com>
+ */
+
+#ifndef __ZEPHYR_RTOS_STRING_MACRO_H__
+#define __ZEPHYR_RTOS_STRING_MACRO_H__
+
+/* treat x as string while forcing x expansion beforehand */
+#define _STRING_QUOTE(x) #x
+#define STRING_QUOTE(x) _STRING_QUOTE(x)
+
+/* concat x and y while forcing x and y expansion beforehand */
+#define _STRING_CONCAT_BASE(x, y) x##y
+#define STRING_CONCAT(x, y) _STRING_CONCAT_BASE(x, y)
```

Comment:
no. This header should just `#include <zephyr/toolchain.h>` and `#include <zephyr/sys/util_macro.h>` and you should just use `STRINGIFY()` and `CONCAT()` directly in SOF code and define **them** in posix and xtos headers.